airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -322,8 +322,26 @@ def execute(self, context: Context):
                     ),
                     method_name="execute_complete",
                 )
+            self._handle_job_error(job)
+            # job.result() returns a RowIterator. Mypy expects an instance of SupportsNext[Any] for
+            # the next() call which the RowIterator does not resemble to. Hence, ignore the arg-type error.
+            records = next(job.result())  # type: ignore[arg-type]
```

Comment:
next call call might returns a row object not row iterator.

Reply:
Yes, it does return only a single row, and we are only checking the first row. I was recreating what's happening in the [Trigger](https://github.com/apache/airflow/blob/420709962ec864fa62b6e1b6c2f723a2c14cb751/airflow/providers/google/cloud/triggers/bigquery.py#L165) itself that checks only the first row (first row is returned in `event["records"]`). I was wondering why it happens as in the execute_complete uses the `all(records):` that does not check much in that case, but that's probably for another issue. For now, i think when not deferred but in deferrable mode, we should recreate what's happening in the trigger to maintain the same behaviour.

Also, this `next()` part is copied from BigQueryValueCheckOperator's [check](https://github.com/apache/airflow/blob/420709962ec864fa62b6e1b6c2f723a2c14cb751/airflow/providers/google/cloud/operators/bigquery.py#L454), so if it's wrong, it should be changed there too.

Reply:
Yes, `next()` returns only a single row, and we are only checking the first row. `validate_records` expect a single row (even though the name of the argument is `records`, plural). This is because I was recreating what's happening in the [Trigger](https://github.com/apache/airflow/blob/420709962ec864fa62b6e1b6c2f723a2c14cb751/airflow/providers/google/cloud/triggers/bigquery.py#L165) itself that checks only the first row (but single row is returned as `event["records"]`). I was wondering why it happens as in the execute_complete uses the `all(records):` that does not check much in that case, but that's probably for another issue. For now, i think when not deferred but in deferrable mode, we should recreate what's happening in the trigger to maintain the same behaviour.

Also, this `next()` part is copied from BigQueryValueCheckOperator's [check](https://github.com/apache/airflow/blob/420709962ec864fa62b6e1b6c2f723a2c14cb751/airflow/providers/google/cloud/operators/bigquery.py#L454), so if it's wrong, it should be changed there too.

Maybe that's not the correct attitude, let me know if there is a better way to do it 😄

Reply:
Well. For me it looks like we should fix it in both places -  both seem wrong,. Seem like in both cases we seem to get over the error accidentally - because single record is a tuple, it's also iterable, so the all() check in "validate_records" and the checks done in "check_value" are iterating over fields in the record not over the records themselves - and it checks for something different than intended.

It seems that (at least that would be logical thing to do here), the right apprach should be to do:

```python
records = [ next(job.result() ]
```

Which would turn the single record into an iterable array (and similar for `check_value`) - but that should of course be verified in some real case, if what we get is what we wanted to get.